src/Illuminate/Support/Testing/Fakes/ExceptionHandlerFake.php

```diff
@@ -0,0 +1,267 @@
+
+    /**
+     * Get the underlying handler implementation.
+     *
+     * @return \Illuminate\Contracts\Debug\ExceptionHandler
+     */
+    public function handler()
+    {
+        return $this->handler;
+    }
+
+    /**
+     * Assert nothing has been reported.
+     *
+     * @return void
+     */
+    public function assertNothingReported()
+    {
+        Assert::assertEmpty(
+            $this->reported,
+            sprintf(
+                'The following exceptions were reported: %s.',
+                implode(', ', array_map('get_class', $this->reported)),
+            ),
+        );
+    }
+
+    /**
+     * Assert if an exception of the given type has been reported.
+     *
+     * @param  \Closure|string  $exception
+     * @return void
+     */
+    public function assertReported(Closure|string $exception)
+    {
+        $message = sprintf(
+            'The expected [%s] exception was not reported.',
+            is_string($exception) ? $exception : $this->firstClosureParameterType($exception)
+        );
+
+        if (is_string($exception)) {
+            Assert::assertTrue(
+                in_array($exception, array_map('get_class', $this->reported), true),
+                $message,
+            );
+
+            return;
+        }
+
+        Assert::assertTrue(
+            collect($this->reported)->contains(
+                fn (Throwable $e) => $this->firstClosureParameterType($exception) === get_class($e)
+                    && $exception($e) === true,
+            ), $message,
+        );
+    }
+
+    /**
+     * Assert the number of exceptions that have been reported.
+     *
+     * @param  int  $count
+     * @return void
+     */
+    public function assertReportedCount(int $count)
+    {
+        $total = collect($this->reported)->count();
+
+        PHPUnit::assertSame(
+            $count, $total,
+            "The total number of exceptions reported was {$total} instead of {$count}."
+        );
+    }
+
+    /**
+     * Assert if an exception of the given type has not been reported.
+     *
+     * @param  \Closure|string  $exception
+     * @return void
+     */
+    public function assertNotReported(Closure|string $exception)
+    {
+        try {
+            $this->assertReported($exception);
+        } catch (ExpectationFailedException $e) {
+            return;
+        }
+
+        throw new ExpectationFailedException(sprintf(
+            'The expected [%s] exception was not reported.',
+            is_string($exception) ? $exception : $this->firstClosureParameterType($exception)
+        ));
+    }
+
+    /**
+     * Report or log an exception.
+     *
+     * @param  \Throwable  $e
+     * @return void
+     */
+    public function report($e)
+    {
+        if ($this->shouldReport($e) && $this->isFakedException($e)) {
+            $this->reported[] = $e;
+
+            if ($this->throwOnReport) {
+                throw $e;
+            }
+        }
+
+        $this->handler->report($e);
+    }
```

Comment:
I feel we should not call `$this->handler->report` if we are faking the exception. This matches other fakes, such as the event fake where we do not dispatch the event if it has been faked.

```suggestion
    /**
     * Report or log an exception.
     *
     * @param  \Throwable  $e
     * @return void
     */
    public function report($e)
    {
        if (! $this->isFakedException($e)) {
            $this->handler->report($e);
            
            return;
        }
        
        if (! $this->shouldReport($e)) {
            return;
        }
        
        $this->reported[] = $e;

        if ($this->throwOnReport) {
            throw $e;
        }
    }
```